packages/integration/src/github/index.ts

```diff
@@ -18,11 +19,21 @@ export const GitHubIntegrationInstalledState = z.object({
 	status: z.literal("installed"),
 	repositories: z.custom<GitHubIntegrationRepository[]>(),
 });
-export const GitHubIntegration = z.discriminatedUnion("status", [
+export const GitHubIntegrationState = z.discriminatedUnion("status", [
 	GitHubIntegrationUnsetState,
 	GitHubIntegrationUnauthorizedState,
 	GitHubIntegrationInvalidCredentialState,
 	GitHubIntegrationNotInstalledState,
 	GitHubIntegrationInstalledState,
 ]);
+export type GitHubIntegrationState = z.infer<typeof GitHubIntegrationState>;
+export const GitHubIntegration = z.object({
+	state: GitHubIntegrationState,
+	components: z
+		.object({
+			authentication: z.custom<ReactNode>(),
+			installation: z.custom<ReactNode>(),
```

Comment:
I'm wondering about the use of ReactNode in the data type. Could you elaborate on the reasoning behind including it?

My concern is that using ReactNode might prevent the schema from being easily serialized (e.g., for storage or transmission). Ideally, we'd like the schema to be composed of serializable values.